app/screens/post_options/options/delete_post_option.tsx

```diff
@@ -1,22 +1,53 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import React from 'react';
+import React, {useCallback} from 'react';
+import {useIntl} from 'react-intl';
+import {Alert} from 'react-native';
 
+import {deletePost} from '@actions/remote/post';
+import {Screens} from '@constants';
+import {useServerUrl} from '@context/server';
 import {t} from '@i18n';
+import {dismissBottomSheet} from '@screens/navigation';
 
 import BaseOption from './base_option';
 
-//fixme: wire up handleDeletePost
-const DeletePostOption = () => {
-    const handleDeletePost = () => null;
+type Props = {
+    postId: string ;
+}
+const DeletePostOption = ({postId}: Props) => {
+    const serverUrl = useServerUrl();
+    const {formatMessage} = useIntl();
+
+    const onPress = useCallback(() => {
+        Alert.alert(
+            formatMessage({id: 'mobile.post.delete_title', defaultMessage: 'Delete Post'}),
+            formatMessage({
+                id: 'mobile.post.delete_question',
+                defaultMessage: 'Are you sure you want to delete this post?',
+            }),
+            [{
+                text: formatMessage({id: 'mobile.post.cancel', defaultMessage: 'Cancel'}),
+                style: 'cancel',
+                onPress: () => dismissBottomSheet(Screens.POST_OPTIONS),
+            }, {
+                text: formatMessage({id: 'post_info.del', defaultMessage: 'Delete'}),
+                style: 'destructive',
+                onPress: () => {
+                    deletePost(serverUrl, postId);
+                    dismissBottomSheet(Screens.POST_OPTIONS);
```

Comment:
Should we dismiss the bottom sheet before the alert? (0/5, more a UX question)

Reply:
yes, the way this works is:
1. User press on the delete option
2. Alert is displayed
3. User presses on `Delete`
4. the delete action is called and the bottom sheet closes.